server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -21,23 +21,32 @@
   protected void doGet(HttpServletRequest req, HttpServletResponse res)
       throws ServletException, IOException {
 
-    String videoName = req.getParameter("name");
+    String input = req.getParameter("name");
 
-    assert videoName == null : "Something went wrong with sending to backend.";
+    assert input == null : "Something went wrong with sending to backend.";
 
     // Use like this: {url_parameter, value}
     HashMap<String, String> videoArgs = new HashMap<>();
     HashMap<String, String> commentArgs = new HashMap<>();
     HashMap<String, String> nameArgs = new HashMap<>();
 
-    videoArgs.put("q", videoName);
-    videoArgs.put("type", "video");
-    String videoIdJson = new YoutubeRequest("search", videoArgs).getResult();
-    String videoId = getVideoId(videoIdJson);
+    String videoId = input; // assume user enters id
 
     commentArgs.put("part", "snippet");
-    commentArgs.put("videoId", videoId);
-    String commentsJson = new YoutubeRequest("commentThreads", commentArgs).getResult();
+    commentArgs.put("videoId", input);
+    String commentsJson;
+    try {
+      commentsJson = new YoutubeRequest("commentThreads", commentArgs).getResult();
+    } catch (IOException err) {
+      // We need to search for the id and get the comments again
+      videoArgs.put("q", input);
+      videoArgs.put("type", "video");
+      String videoIdJson = new YoutubeRequest("search", videoArgs).getResult();
+      videoId = getVideoId(videoIdJson);
+
+      commentArgs.replace("videoId", videoId);
+      commentsJson = new YoutubeRequest("commentThreads", commentArgs).getResult();
+    }
```

Comment:
Nick will prob say something about how doing try/catch is lazy/bad practice. Could you catch the error by adding an if statement to check whether a variable is null?

Reply:
`try`/`catch` is not bad practice. Why would it be?

Also, he's catching an `IOException`. That has nothing to do with whether any variable is `null`.

Reply:
+1. `try/catch` is only bad when you're using it as a cure-all to catch errors in your code (null pointer exceptions, divide by zero errors, etc). It's necessary when interacting with a function that explicitly throws errors, such as `IOException`'s. It's generally best practice to minimize the surface area of your `try/catch`, which is being done here.